libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using prompts = Microsoft.Bot.Builder.Prompts;
+using static Microsoft.Bot.Builder.Prompts.PromptValidatorEx;
+using Microsoft.Bot.Schema;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Prompts;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    public class OAuthPromptSettingsWithTimeout : prompts.OAuthPromptSettings
+    {
+        public int? Timeout { get; set; }
+    }
+
+    public class OAuthPromptState : PromptOptions
+    {
+        public OAuthPromptState(PromptOptions defaultPromptOptions) : base()
+        {
+            if (defaultPromptOptions != null)
+            {
+                PromptString = PromptString ?? defaultPromptOptions.PromptString;
+                PromptActivity = PromptActivity ?? defaultPromptOptions.PromptActivity;
+                Speak = Speak ?? defaultPromptOptions.Speak;
+                RetryPromptString = RetryPromptString ?? defaultPromptOptions.RetryPromptString;
+                RetryPromptActivity = RetryPromptActivity ?? defaultPromptOptions.RetryPromptActivity;
+                RetrySpeak = RetrySpeak ?? defaultPromptOptions.RetrySpeak;
+            }
+        }
+
+        public DateTime Expires { get; set; }
+    }
+
+    public class OAuthPrompt : Control, IDialog, IDialogContinue
+    {
+        private prompts.OAuthPrompt _prompt;
```

Comment:
This is confusing to me. 

This class is OAuthPrompt. Why is there a member variable of the same type as the class? The only time I normally see things like that is Singleton and related patterns. Can you explain?

Reply:
there is an oauthprompt class in the prompt and the dialogs namespace. the dialogs namespace prompt refers the one in the prompts namespace. i see this design for other prompts as well. forexample, look at choiceprompt

---
In reply to: [184818382](https://github.com/Microsoft/botbuilder-dotnet/pull/520#discussion_r184818382) [](ancestors = 184818382)